mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1328,6 +1328,16 @@ LogicalResult CppEmitter::emitAttribute(Location loc, Attribute attr) {
     }
   };
 
+  auto getDenseElementType = [&](const Type &type) {
+    if (auto arrayType = dyn_cast<ArrayType>(type)) {
+      return arrayType.getElementType();
+    }
+    if (auto tensorType = dyn_cast<TensorType>(type)) {
+      return tensorType.getElementType();
+    }
+    return Type();
+  };
+
```

Comment:
Should this be `ShapedType` instead? Also, it might be worth considering adding an `assert`

Reply:
Thanks for comment, changed to use `ShapedType` and added assert.